lib/interferon.rb

```diff
@@ -133,12 +133,55 @@ def read_hosts(sources)
       return hosts
     end
 
-    def update_alerts(destinations, hosts, alerts, groups)
+    def update_alerts(destinations, hosts, alerts, groups, dry_run)
       loader = DestinationsLoader.new([@alerts_repo_path])
       loader.get_all(destinations).each do |dest|
         break if @request_shutdown
         log.info "updating alerts on #{dest.class.name}"
-        update_alerts_on_destination(dest, hosts, alerts, groups)
+        if dry_run
+          dry_run_update_alerts_on_destination(dest, hosts, alerts, groups)
+        else
+          update_alerts_on_destination(dest, hosts, alerts, groups)
+        end
+      end
+    end
+
+    def dry_run_update_alerts_on_destination(dest, hosts, alerts, groups)
```

Comment:
In general I think you are right. The issue I am not doing this is that the logic for dry-run and non dry-run are quite different. I tried to avoid long methods with lots of condition checks. 
Here are more details:
The common part of the updating logics between dry run and regular run is essentially only building of the alerts queue from alerts files, after this the logic splits:
- For dry-run, I compare the queue against existing alerts in datadog and reject all existing ones, then change alerts names, call datadog api to create them. Afterwards I call datadog api to delete the newly created together with any previously failed to delete dry run alerts.

- For regular run(previous existing logic), alerts in alerts queue are the ones to create or update, then a comparison is done against existing alerts in datadog, extra ones are deleted.